pkg/processor/trigger/batcher.go

```diff
@@ -50,36 +50,44 @@ func (b *Batcher) Add(event nuclio.Event, responseChan *common.ChannelWithRecove
 	b.currentBatch <- &BatchedEventWithResponse{event: event, responseChan: responseChan}
 
 	// if batchIsFull, Write to `batchIsFull` chan, so that we send batch to worker right when batch len reached the maximum
-	if cap(b.currentBatch) == len(b.currentBatch) {
+	// plus one, because we read the first event in WaitForBatch separately
+	if cap(b.currentBatch)+1 == len(b.currentBatch) {
```

Comment:
I think it should be `-1` instead of `+1` : 
You define a batch size (e.g. 10), so `cap` is 10.
You then read 1 event, so the batch is full when there are 9 items in the current batch = cap-1

Reply:
@TomerShor absolutely right, it should be +1 to len, not to capacity!

Reply:
actually now it should be `>` because it is possible that both goroutines are waiting on adding to the channel